src/agent/communicator/include/https_socket.hpp

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <ihttp_socket.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/asio/experimental/awaitable_operators.hpp>
+#include <boost/asio/ssl.hpp>
+#include <boost/beast.hpp>
+#include <boost/system/error_code.hpp>
+
+#include <string>
```

Comment:
Probably awaitable_operators can go to the source file, not sure about the rest, but worth to take a look.

Reply:
Done

Reply:
was this change pushed?

Reply:
mmm... This is old, I might have added some headers back again sometime today while testing something else...